quill/api/src/game.rs

```diff
@@ -97,6 +98,15 @@ impl Game {
         EntityBuilder::new(id)
     }
 
+    /// Spawns a new [`Entity`] with the supllied components.
+    ///
+    /// The builder is not initialized with any default components.
+    pub fn spawn(&self, components: impl DynamicBundle) -> Entity {
```

Comment:
Could we rename this to `spawn_entity_bundle` or similar? We already have `spawn_entity`, which is higher level and probably more frequently used.

Reply:
What if I make `spawn` more generic, and make all components `DynamicBundle` and make the `tuple_impl` over `DynamicBundle`. This way `spawn_entity` would become `spawn(DefaultBundle)` or with more data attached `spawn((DefaultBundle::new(...), Health::default())`

Reply:
`DefaultBundle` would have to be special cased in the server if we do this. There are a number of internal components inside the server that aren't exposed to plugins, so plugins aren't directly able to add all the default components for an entity.

Reply:
I was not aware of this, the DefaultBundle have full access to the EntityBuilder so are there no way we can handle it in Quill?

Reply:
EntityBuilder in Quill can only access components available to plugins, but many components aren't exposed to plugins, because they're implementation details. (`SpawnPacket`, `NetworkId`, a few others). Also, we might want to add new internal components in the future, and hardcoding the default component sets in plugins would make this non-backwards-compatible. In short, the server needs to be aware of the type of entity we're creating so it can initialize default components.